integration/querier_test.go

```diff
@@ -410,6 +414,170 @@ func TestQuerierWithBlocksStorageRunningInSingleBinaryMode(t *testing.T) {
 			} else if testCfg.indexCacheBackend == tsdb.IndexCacheBackendMemcached {
 				require.NoError(t, cluster.WaitSumMetrics(e2e.Equals(float64((11+2)*seriesReplicationFactor)), "thanos_memcached_operations_total")) // as before + 2 gets
 			}
+
+			// Query metadata.
+			testMetadataQueriesWithBlocksStorage(t, c, series1, series2, series3, blockRangePeriod)
+		})
+	}
+}
+
+func testMetadataQueriesWithBlocksStorage(t *testing.T, c *e2ecortex.Client, series1, series2, series3 []prompb.TimeSeries, blockRangePeriod time.Duration) {
```

Comment:
Nit: I've found it bit confusing to express `from` and `to` in terms of series, and would suggest passing three points in time instead, relevant to the test:

- minT of oldest block in ingester
- minT of head
- maxT of head

Reply:
Reason I passed the timeseries is because we also need labels (eg. `series3[0].Labels`)